texar/modules/encoders/xlnet_encoders.py

```diff
@@ -0,0 +1,530 @@
+            memory: Memory from previous batches. A list of length `num_layers`,
+                each tensor of shape `[batch_size, mem_len, hidden_dim]`.
+            permute_mask: The permutation mask. Float tensor of shape
+                `[batch_size, seq_len, seq_len]`.
+                A value of 0 for ``permute_mask[i, j, k]`` indicates that
+                position `i` attends to position `j` in batch `k`.
+            target_mapping: The target token mapping. Float tensor of shape
+                `[batch_size, num_targets, seq_len]`.
+                A value of 1 for ``target_mapping[i, j, k]`` indicates that
+                the `i`-th target token (in order of permutation) in batch `k`
+                is the token at position `j`.
+                Each row ``target_mapping[i, :, k]`` can have no more than one
+                value of 1.
+            bi_data (bool): Whether to use bidirectional data input pipeline.
+            clamp_len (int): Clamp all relative distances larger than
+                :attr:`clamp_len`. A value of -1 means no clamping.
+            cache_len (int): Length of memory (number of tokens) to cache.
+            same_length (bool): Whether to use the same attention length for
+                each token.
+            attn_type (str): Attention type. Supported values are `"uni"`
+                and `"bi"`.
+            two_stream (bool): Whether to use two-stream attention. Only set to
+                `True` when pre-training or generating text. Defaults to
+                `False`.
+
+        :returns: A tuple of `(output, new_memory)`:
+
+            - **`output`**: The final layer output representations. Shape
+              `[batch_size, seq_len, hidden_dim]`.
+            - **`new_memory`**: The memory of the current batch.
+              If `cache_len` is 0, then `new_memory` is `None`. Otherwise, it is
+              a list of length `num_layers`, each tensor of shape
+              `[batch_size, cache_len, hidden_dim]`.
+              This can be used as the :attr:`memory` argument in the next batch.
+        """
+        return self._execute(self.word_embedder(token_ids),
+                             segment_ids=segment_ids, input_mask=input_mask,
+                             memory=memory, permute_mask=permute_mask,
+                             target_mapping=target_mapping, bi_data=bi_data,
+                             clamp_len=clamp_len, cache_len=cache_len,
+                             same_length=same_length, attn_type=attn_type,
+                             two_stream=two_stream, **kwargs)
+
+    def _execute(self, word_embed, segment_ids=None, input_mask=None,
+                 memory=None, permute_mask=None, target_mapping=None,
+                 bi_data=False, clamp_len=None, cache_len=0, same_length=False,
+                 attn_type='bi', two_stream=False, mode=None):
+        r"""Compute XLNet representations for the input.
```

Comment:
If the docstring is the same as that of forward, we can simply say this and omit the details. Or, if it's mostly the same, just point out the differences.

FYI, the pytorch PR https://github.com/asyml/texar-pytorch/pull/128 fixed this.

Reply:
Changes are fixed in f578727